### PR TITLE
Add Opt-in support to new meta tests

### LINKED ISSUE #103: make new common tests warnings until repos have opted-in
There should be a file in each repo that says which common tests the repo has passed.  Those are treated as an error.  Once someone gets that test passing they add the test to the file so that regressions will cause a failure.

### LINKED ISSUE #104: Once repos have fixed example make the 'validate example files' test a failure
This is related to #103

### PR DESCRIPTION
**Pull request by TravisEz13**
Fixes #103
Fixes #104 in a way

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/dscresource.tests/105)

### CODE REVIEW

**Comment by TravisEz13** [comment]
@PlagueHO @johlju @kwirkykat I think this is a pattern we can use to phase in Meta tests without breaking everything.  
I tested it here: https://ci.appveyor.com/project/TravisEz13/xwindowsupdate/build/2.5.115.0

**Comment by PlagueHO** [comment]
@TravisEz13 - that looks pretty good to me.

What might be good would be to include a copy of the ```.MetaTestOptIn.json``` file in the DSCResource.tests with everything "opted in". That way we could just copy it to a repo and delete the tests to opt out of.

**Comment by johlju** [comment]
@TravisEz13 This works for me. 

Just want to clarify the opt-in.

If we opt-in, wouldn't we do that by changing the appveyor.yml to the new layout? Using the new Invoke-* functions.
Example of opt-out of markdown and example tests: https://github.com/PowerShell/xSQLServer/blob/dev/appveyor.yml

Using this will by default opt-out of markdown and example test.
```
Invoke-AppveyorTestScriptTask -CodeCoverage
```
or this

```
Invoke-AppveyorTestScriptTask -CodeCoverage -ExcludeTag 'Examples','Markdown'
```

Using this will opt-out of markdown test only
```
Invoke-AppveyorTestScriptTask -CodeCoverage -ExcludeTag 'Markdown'
```

Using this will **opt-in for all of the tests**
```
Invoke-AppveyorTestScriptTask -CodeCoverage -ExcludeTag @()
```

**Comment by TravisEz13** [comment]
@johlju @PlagueHO I understand.  This should only be an issue until all repo's update to the new pattern of calling the common tests but that could be some time and there are a lot of repos.

**Comment by johlju** [comment]
@TravisEz13 Yep, I'm with you. This change will fix all repo's until then. So when we add the new pattern to the repo to opt-in for the tests, we would also need to add the file `.MetaTestOptIn.json` with the correct value to opt-in for actually get the test failing. This works for me.

File: Meta.Tests.ps1 [hunk 1]
```diff
@@ -36,6 +36,19 @@ if (-not $repoRootPathFound)
     $repoRootPath = $moduleRootFilePath
 }
 $repoName = Split-Path -Path $repoRootPath -Leaf
+$testOptInFilePath = Join-Path $repoRootPath '.MetaTestOptIn.json'
```
Comment (johlju) [inline Meta.Tests.ps1]: No positional parameters

File: TestHelper.psm1 [hunk 2]
```diff
@@ -825,6 +825,69 @@ function Install-NugetExe
...
+#>
+function Get-PesterDescribeOptInStatus
+{
+    param
+    (
+        [string[]]$OptIns
+    )
+
+    $describeName = Get-PesterDescribeName
+    $optin = $OptIns -icontains $describeName
```
Comment (johlju) [inline TestHelper.psm1]: $optIn (upper 'I')  - Since you write the other as `$OptIns`
Same for all occurences

**Comment by johlju** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresource.tests/105#-:-KcT6AsVB0C92jBB3e-z:bnfp4nl)*

**Comment by kwirkykat** [comment]
@TravisEz13 I'm ok with this as a temporary solution until all the appveyor.yml's across the repositories are updated.

**Pull request merged**